src/Raven.Server/Documents/ETL/Providers/Queue/QueueBrokerConnectionHelper.cs

```diff
@@ -80,4 +88,65 @@ public static IConnection CreateRabbitMqConnection(RabbitMqConnectionSettings se
         var connectionFactory = new ConnectionFactory { Uri = new Uri(settings.ConnectionString) };
         return connectionFactory.CreateConnection();
     }
+
+    public static QueueClient CreateAzureQueueStorageClient(
+        AzureQueueStorageConnectionSettings azureQueueStorageConnectionSettings, string queueName)
+    {
+        QueueClient queueClient;
+
+        if (azureQueueStorageConnectionSettings.Authentication.ConnectionString != null)
+        {
+            queueClient = new QueueClient(azureQueueStorageConnectionSettings.Authentication.ConnectionString,
+                queueName);
+        }
+
+        else if (azureQueueStorageConnectionSettings.Authentication.EntraId != null)
+        {
+            var queueUri = new Uri($"{azureQueueStorageConnectionSettings.GetStorageUrl()}{queueName}");
+
+            queueClient = new QueueClient(
+                queueUri,
+                new ClientSecretCredential(
+                    azureQueueStorageConnectionSettings.Authentication.EntraId.TenantId,
+                    azureQueueStorageConnectionSettings.Authentication.EntraId.ClientId,
+                    azureQueueStorageConnectionSettings.Authentication.EntraId.ClientSecret));
+        }
+        else if(azureQueueStorageConnectionSettings.Authentication.Passwordless)
+        {
+            var queueUri = new Uri($"{azureQueueStorageConnectionSettings.GetStorageUrl()}{queueName}");
+            queueClient = new QueueClient(queueUri, new DefaultAzureCredential());
+        }
+        else
+        {
+            throw new NotSupportedException("Provided authentication method is not supported");
```

Comment:
Please make sure it will be validate much earlier - on attempt to put a connection string. We should never get here

Reply:
Just removed this part. You can not create auth option which doesn't exist in AzureQueueStorageConnectionSettings